src/FileMigrationHelper.php

```diff
@@ -102,6 +102,17 @@ protected function migrateFile($base, File $file, $legacyFilename)
             return false;
         }
 
+        // If this file has the wrong class for it's extension, update the classname
+        $destinationClass = $file->get_class_for_file_extension($extension);
+        if ($file->ClassName !== $destinationClass) {
+            // We disable validation so that we are able to write a corrected
+            // classname, once that is changed we re-enable it for subsequent writes
+            Config::modify()->set(File::class, 'validation_enabled', false);
```

Comment:
We probably should read the value of `validation_enabled` before setting it `false`. When we're done doing things that require validation to be false, then we reset it to its original value.

The way it's done right now, if someone decides to disable this validation, we'll end up enabling it for them by accident.

Reply:
Done